amun/core.py

```diff
@@ -114,10 +124,19 @@ def create_inspect_job(openshift: OpenShift, image_stream_name: str, specificati
 
     if 'run' in specification:
         run_specification = specification['run']['requests']
+        hardware_specification = specification['run']['hardware']
```

Comment:
Here you expect "hardware" key to be always set. I think it will be good to keep hardware information optional.

Reply:
Made "requests" and "hardware" during build and run time optional